lib/src/native/realm_core.dart

```diff
@@ -81,7 +81,7 @@ class _RealmCore {
 
     String? message;
     if (error.ref.message != nullptr) {
-      message = error.ref.message.cast<Utf8>().toDartString();
+      message = error.ref.message.cast<Utf8>().toRealmDartString();
```

Comment:
we are not reaping the benefit of introducing `toRealmDartString` here. I think the previous version was removing the null check

Reply:
yeah, I overlooked that